packages/@react-native-windows/cli/src/utils/architecturePrompt.ts

```diff
@@ -0,0 +1,142 @@
+        (resolve, reject) => {
+          prompts(
+            {
+              type: 'text',
+              name: 'choice',
+              message:
+                'Would you like to continue using the Old Architecture? (Y/N)',
+              validate: (value: string) => {
+                const normalized = value.trim().toLowerCase();
+                return normalized === 'y' || normalized === 'n'
+                  ? true
+                  : "Invalid input. Please enter 'Y' for Yes or 'N' for No.";
+              },
+            },
+            {
+              onCancel: () => reject(new Error('User cancelled')),
+            },
+          )
+            .then(resolve)
+            .catch(reject);
+        },
+      );
+
+      // Timeout fallback with clearTimeout support
+      const timeoutPromise = new Promise<{choice?: string}>(resolve => {
+        timeoutId = setTimeout(() => {
+          console.log(
+            chalk.yellow(
+              '\n⏳ No input received in 3 seconds. Proceeding with Old Architecture by default.',
+            ),
+          );
+          resolve({choice: 'y'});
+        }, 3000);
+      });
+
+      const response = await Promise.race([userInputPromise, timeoutPromise]);
+
+      // ensures timeout callback never runs after user input
+      if (timeoutId) clearTimeout(timeoutId);
+
+      if (!response.choice) {
+        // User cancelled or no input
+        return {shouldContinueWithOldArch: true, userCancelled: true};
+      }
+
+      const normalizedChoice = response.choice.trim().toLowerCase();
+
+      if (normalizedChoice === 'y') {
+        console.log(
+          chalk.yellow(
+            'Proceeding with Old Architecture. You can migrate later using our migration guide: https://microsoft.github.io/react-native-windows/docs/new-architecture',
+          ),
+        );
+        return {shouldContinueWithOldArch: true, userCancelled: false};
+      } else if (normalizedChoice === 'n') {
+        console.log(
+          chalk.green(
+            'Great choice! Setting up the project with New Architecture support.',
+          ),
+        );
+        return {shouldContinueWithOldArch: false, userCancelled: false};
+      }
+    } catch (error) {
+      if ((error as Error).message === 'User cancelled') {
+        console.log(
+          chalk.yellow(
+            '\nNo input received. Proceeding with Old Architecture by default. You can opt into the New Architecture later.',
+          ),
+        );
+        return {shouldContinueWithOldArch: true, userCancelled: true};
+      }
+
+      // retry on invalid input
+      attempts++;
+      if (attempts < maxRetries) {
+        console.log(
+          chalk.red("Invalid input. Please enter 'Y' for Yes or 'N' for No."),
+        );
+      }
+    }
+  }
+
+  // Max retries reached
+  console.log(
+    chalk.yellow(
+      `\nMax retries reached. Proceeding with Old Architecture by default. You can opt into the New Architecture later.`,
```

Comment:
There's no need for max retry logic. The user either answers yes or no, and if they never do, there's no reason to just go forward.